coded_tools/deep_rag/create_networks.py

```diff
@@ -0,0 +1,355 @@
+                to the coded tool and whose values are the values passed for them
+                by the calling agent.  This dictionary is to be treated as read-only.
+        :param sly_data: A dictionary whose keys are defined by the agent hierarchy,
+                but whose values are meant to be kept out of the chat stream.
+
+                This dictionary is largely to be treated as read-only.
+                It is possible to add key/value pairs to this dict that do not
+                yet exist as a bulletin board, as long as the responsibility
+                for which coded_tool publishes new entries is well understood
+                by the agent chain implementation and the coded_tool implementation
+                adding the data is not invoke()-ed more than once.
+        :return: A return value that goes into the chat stream.
+        """
+        # Get most args as members
+        empty: Dict[str, Any] = {}
+        self.grouping_json: Dict[str, Any] = args.get("grouping_json", empty)
+        self.files_directory: str = args.get("files_directory")
+
+        logstr: str = dumps(self.grouping_json, indent=4, sort_keys=True)
+        self.logger.info("grouping_json is %s", logstr)
+
+        reservationist: Reservationist = args.get("reservationist")
+        deployments: Dict[Reservation, Dict[str, Any]] = await self.assemble_deployments(reservationist)
+
+        # Deploy the reservations with confirmation event
+        # If you don't really need to wait until the new agent(s) has been deployed
+        # then set confirmation=False, and don't bother about waiting for the Event.
+        deployed_event: Event = None
+        try:
+            async with reservationist:
+                deployed_event = await reservationist.deploy(deployments, confirmation=True)
+
+        except ValueError as exception:
+            # Report exceptions from below as errors here.
+            error: str = f"{exception}"
+            self.logger.error(error)
+            return error
+
+        if deployed_event is not None:
+            await deployed_event.wait()
+
+        # Assemble the output
+        reservation_info: List[Dict[str, Any]] = self.assemble_reservation_info(deployments.keys())
+        sly_data["agent_reservations"] = reservation_info
+        sly_data["aa_grouping_json"] = self.grouping_json
+
+        entry: Dict[str, Any] = reservation_info[-1]
+        entry_reservation_id: str = entry.get("reservation_id")
+        entry_lifetime: str = entry.get("lifetime_in_seconds")
+        output: str = f"The main agent to access your deep rag network is {entry_reservation_id}" + \
+                      f"Hurry, it's only available for {entry_lifetime} seconds."
+        return output
+
+    async def assemble_deployments(self, reservationist: Reservationist) -> Dict[Reservation, Dict[str, Any]]:
+        """
+        Create all the networks that are to be deployed together.
+        """
```

Comment:
Main workhorse method that gets together the contents of all the networks for the deep rag, given the groupings of files from the calling deep_rag.hocon network.